pilot/pkg/bootstrap/server.go

```diff
@@ -439,7 +439,8 @@ func (s *Server) initMeshNetworks(args *PilotArgs) error {
 		return nil
 	}
 	log.Infof("mesh networks configuration %s", spew.Sdump(meshNetworks))
-	util.ResolveHostsInNetworksConfig(s.meshNetworks)
+	util.ResolveHostsInNetworksConfig(meshNetworks)
```

Comment:
So before we were resolving the hosts for the wrong meshNetworks config? Good catch